application/frontend/controllers/ResetController.php

```diff
@@ -81,9 +128,98 @@ public function actionResend($uid)
             throw new NotFoundHttpException();
         }
 
+        /*
+         * Resend verification
+         */
         $reset->send();
 
-        \Yii::$app->response->statusCode = 204;
-        return;
+        return $reset;
+    }
+
+    /**
+     * Validate reset code. Logs user in if successful
+     * @param string $uid
+     * @return Reset
+     * @throws BadRequestHttpException
+     * @throws NotFoundHttpException
+     * @throws ServerErrorHttpException
+     * @throws \Exception
+     */
+    public function actionValidate($uid)
+    {
+        /** @var Reset $reset */
+        $reset = Reset::findOne(['uid' => $uid]);
+        if ($reset === null) {
+            throw new NotFoundHttpException();
+        }
+
+        $code = \Yii::$app->request->getBodyParam('code', null);
+        if ($code === null) {
+            throw new BadRequestHttpException('Code is required', 1462989866);
+        }
+
+        $log = [
+            'action' => 'Validate reset',
+            'reset_id' => $reset->id,
+        ];
+
+        $isValid = $reset->isUserProvidedCodeCorrect($code);
+        if ($isValid === true) {
+
+            EventLog::log(
+                'ResetVerificationSuccessful',
+                [
+                    'reset_id' => $reset->id,
+                    'type' => $reset->type,
+                    'attempts' => $reset->attempts,
+                ],
+                $reset->user_id
+            );
+
+            /*
+             * Reset verified successfully, log user in
+             */
+            if (\Yii::$app->user->login($reset->user, \Yii::$app->params['sessionDuration'])) {
+                $log['status'] = 'success';
+                \Yii::warning($log);
+
+                /*
+                 * Delete reset record, log errors, but let user proceed
+                 */
+                if ( ! $reset->delete()) {
+                    \Yii::error([
+                        'action' => 'delete reset after validation',
+                        'reset_id' => $reset->id,
+                        'status' => 'error',
+                        'error' => Json::encode($reset->getFirstErrors()),
+                    ]);
+                }
+
+                return new \stdClass();
```

Comment:
fixed the phpdoc. I return a stdClass so it returns an empty object and 200 status